storage/aptosdb/src/pruner/mod.rs

```diff
@@ -54,8 +54,8 @@ pub(crate) struct Pruner {
     #[allow(dead_code)]
     min_readable_version: Arc<Mutex<Vec<Version>>>,
     /// We send a batch of version to the underlying pruners for performance reason. This tracks the
-    /// last version we sent to the pruner.
-    last_version_sent_to_pruners: Arc<Mutex<Version>>,
+    /// last version we sent to the pruners. If both of the pruners are disabled (i.e. state_store_prune_window and  ledger_prune_window are None,) this value will be None. Otherwise, this value represents the last version we sent to either the state_store_pruner or the ledger_pruner or both.
+    last_version_sent_to_pruners: Option<Arc<Mutex<Version>>>,
```

Comment:
A side note: I guess we might eventually want to maintain two versions separately, given the state commit will happen in async thread. WDYT @lightmark @msmouse ?

Reply:
True, I'd prefer making them completely separate.. I can even imagine the state pruner being ran inside of the asynchronous state snapshot saving thread. -- no write contention on the state tree db at all.

Reply:
(But yeah it's irrelevant on this PR.)

Reply:
Yes we won't call `maybe_wake_pruner` if both are disabled but conceptually this value should be None if both are disabled so I make it optional. It does not seem to add much complexity to the code. `last_version_sent_to_pruners` is only used in a test for the time being.

Reply:
`last_version_sent_to_pruners` is being used outside of tests as well IIUC.  IMO, it doesn't make sense to have this as an option as `maybe_wake_pruner` where this is being used will never be called if pruning is disabled. In fact, by making this `Optional` we have to deal with ugly `unwarp` in `maybe_wake_pruner` function. How do we feel about leaving it as is instead?

Reply:
SG. Reverted the change.